lib/oban/plugins/cron.ex

```diff
@@ -107,9 +122,32 @@ defmodule Oban.Plugins.Cron do
   def handle_info(:evaluate, %State{} = state) do
     state = schedule_evaluate(state)
 
-    Query.with_xact_lock(state.conf, state.lock_key, fn ->
-      insert_jobs(state.conf, state.crontab, state.timezone)
-    end)
+    start_metadata = %{
+      config: state.conf,
+      crontab: state.crontab,
+      name: state.name,
+      interval: state.interval
+    }
+
+    :telemetry.span(
+      [:oban, :plugin, :cron],
+      start_metadata,
+      fn ->
+        state.conf
+        |> Query.with_xact_lock(state.lock_key, fn ->
+          insert_jobs(state.conf, state.crontab, state.timezone)
+        end)
+        |> case do
+          {:ok, inserted_jobs} when is_list(inserted_jobs) ->
+            metadata = %{jobs: extract_jobs_from_insert(inserted_jobs)}
```

Comment:
The other plugins (pruner, stager) don't load the jobs. For the pruner that is impossible and for the stager it's kind of wasteful. They don't need to match necessarily, but what do you think about having the complete jobs here and only counts for pruning/staging?

Reply:
That's actually the approach I was going to take hah. Great minds think alike :P.